source/libtomo/accel/CMakeLists.txt

```diff
@@ -1,62 +1,73 @@
+                                    gpu/utils.cu)
 
-    list(APPEND ACCEL_LIBRARY_SOURCES ${CUDA_LIBRARY_SOURCES})
+  target_include_directories(tomo-accel
+                             PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
 
-    list(APPEND TARGET_INCLUDE_DIRECTORIES
-        ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
-
-endif(TOMOPY_USE_CUDA)
+  target_link_libraries(tomo-accel PRIVATE tomopy::cuda-npp)
 
-add_library(libtomopy-accel SHARED ${ACCEL_LIBRARY_SOURCES})
+  set(CUDA_PROPERTIES
+      CUDA_STANDARD
+      ${CMAKE_CXX_STANDARD}
+      CUDA_STANDARD_REQUIRED
+      ON
+      CUDA_SEPARABLE_COMPILATION
+      ON
+      CUDA_RESOLVE_DEVICE_SYMBOLS
+      ON)
 
-target_link_libraries(libtomopy-accel
-    PUBLIC tomopy::cuda-npp ${TOMOPY_EXTERNAL_LIBRARIES}
-    PRIVATE ${TOMOPY_EXTERNAL_PRIVATE_LIBRARIES})
+  set_target_properties(tomo-accel PROPERTIES ${CUDA_PROPERTIES})
 
-set(CUDA_PROPERTIES
-    CUDA_STANDARD               ${CMAKE_CXX_STANDARD}
-    CUDA_STANDARD_REQUIRED      ON
-    CUDA_SEPARABLE_COMPILATION  ON
-    CUDA_RESOLVE_DEVICE_SYMBOLS ON)
-
-list(APPEND TARGET_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}")
-list(APPEND TARGET_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/../../include/accel")
-
-# target properties
-set_target_properties(libtomopy-accel
-    PROPERTIES
-    PREFIX ""
-    RUNTIME_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    LIBRARY_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    ARCHIVE_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    INCLUDE_DIRECTORIES         "${TARGET_INCLUDE_DIRECTORIES}"
-    C_STANDARD                  ${CMAKE_C_STANDARD}
-    C_STANDARD_REQUIRED         ${CMAKE_C_STANDARD_REQUIRED}
-    CXX_STANDARD                ${CMAKE_CXX_STANDARD}
-    CXX_STANDARD_REQUIRED       ${CMAKE_CXX_STANDARD_REQUIRED}
-    ${CUDA_PROPERTIES})
+endif(TOMOPY_USE_CUDA)
 
-target_compile_definitions(libtomopy-accel PUBLIC
-    ${${PROJECT_NAME}_DEFINITIONS})
+target_link_libraries(tomo-accel PRIVATE ${TOMOPY_EXTERNAL_LIBRARIES}
+                                         ${TOMOPY_EXTERNAL_PRIVATE_LIBRARIES})
 
-target_compile_options(libtomopy-accel PUBLIC
-    $<$<COMPILE_LANGUAGE:C>:${${PROJECT_NAME}_C_FLAGS}>
-    $<$<COMPILE_LANGUAGE:CXX>:${${PROJECT_NAME}_CXX_FLAGS}>
-    $<$<COMPILE_LANGUAGE:CUDA>:${${PROJECT_NAME}_CUDA_FLAGS}>)
+target_compile_definitions(tomo-accel PRIVATE ${${PROJECT_NAME}_DEFINITIONS})
 
-install(TARGETS libtomopy-accel
-    DESTINATION ${PYTHON_SITE_DIR}/tomopy/util/extern
-    COMPONENT python)
+target_compile_options(
+  tomo-accel
+  PUBLIC $<$<COMPILE_LANGUAGE:C>:${${PROJECT_NAME}_C_FLAGS}>
+         $<$<COMPILE_LANGUAGE:CXX>:${${PROJECT_NAME}_CXX_FLAGS}>
+         $<$<COMPILE_LANGUAGE:CUDA>:${${PROJECT_NAME}_CUDA_FLAGS}>)
 
+install(TARGETS tomo-accel EXPORT libtomoTargets)
 
+install(
+  FILES ${HEADERS}
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libtomo
+  COMPONENT Headers)
 
+list(APPEND libtomo_TARGETS tomo-accel)
```

Comment:
So this won't work. Local variables are scoped to directories. `libtomo_TARGETS` in the parent directory will be unaffected by this append.

Replace all your `add_library` calls with `tomopy_add_library` which will just be a thin wrapper around it:

```cmake
function(TOMOPY_ADD_LIBRARY _TARGET)
    # basically this makes tomopy_add_library(...) act exactly like add_library(...)
    add_library(${_TARGET} ${ARGN})

    # check the name has the "tomo-" prefix (expected by libtomoConfig.cmake)
    if(NOT "${_TARGET}" MATCHES "^(tomo-)")
        message(FATAL_ERROR 
            "The name of the target passed to tomopy_add_library MUST start with 'tomo-'")
    endif()

    # create an alias library in the build tree. This allows parent projects adding 
    # tomopy as a submodule to be able to always reference the libtomo targets 
    # with the libtomo:: namespace
    add_library(libtomo::${_TARGET} ALIAS ${_TARGET})

    # append to global property. Use the namespaced name because when 
    # libtomoConfig.cmake imports the targets tomo-accel will not be the name
    # of the target, libtomo::tomo-accel will be the name
    set_property(GLOBAL APPEND PROPERTY libtomo_TARGETS "libtomo::${_TARGET} ")
endfunction()
```

(just define this function in the `MacroUtilities.cmake` file)

Side note: you'll probably want to add this to the `.cmake-format.yml`:

```yaml
parse:
  additional_commands:
    tomopy_add_library:
      flags:
      - STATIC
      - SHARED
      - MODULE
      - EXCLUDE_FROM_ALL
```

to get it to format the same as `find_library`

Reply:
Added this config and formatted more files